tests/test_otel.py

```diff
@@ -0,0 +1,349 @@
+import binascii
+import niquests
+import pytest
+import socket
+import time
+import urllib
+import urllib3
+
+
+NGINX_CONFIG = """
+{{ globals }}
+
+daemon off;
+
+events {
+}
+
+http {
+    {{ http_globals }}
+
+    ssl_certificate localhost.crt;
+    ssl_certificate_key localhost.key;
+
+    otel_exporter {
+        endpoint 127.0.0.1:14317;
+        interval {{ interval or "1ms" }};
+        batch_size 3;
+        batch_count 3;
+    }
+
+    otel_trace on;
+    otel_service_name test_service;
+
+    server {
+        listen       127.0.0.1:18443 ssl;
+        listen       127.0.0.1:18443 quic;
+        listen       127.0.0.1:18080;
+
+        http2 on;
+
+        server_name  localhost;
+
+        location /ok {
+            otel_span_name ok_location;
```

Comment:
As was already suggested, this make more sense in "/custom" location together with other "custom" things. Keeping default here would also make test code more straightforward.

Reply:
Make sense, moved to custom.